shared/src/price_estimate.rs

```diff
@@ -128,13 +132,25 @@ impl UniswapPriceEstimator {
             |_, path, pools| estimate_spot_price(path, pools),
         )
         .await
-        .and_then(|(path, price)| {
-            Ok((
-                path,
-                big_rational_to_float(price)
-                    .ok_or_else(|| anyhow!("Cannot convert price ratio to float"))?,
-            ))
-        })
+    }
+
+    /*
+        Returns a vector of (rational) prices for the given tokens
+        denominated in denominator_token.
+    */
+    pub async fn best_execution_spot_prices(
+        &self,
+        tokens: &[H160],
+        denominator_token: H160,
+    ) -> Result<Vec<BigRational>> {
+        let res = join_all(
+            tokens
+                .iter()
+                .map(|token| self.best_execution_spot_price(*token, denominator_token)),
+        )
+        .await;
+        let res: Result<Vec<(Vec<H160>, BigRational)>, anyhow::Error> = res.into_iter().collect();
```

Comment:
nit: maybe document that this method will return an error as soon as a single estimate fails (it's somewhat logical looking at the signature, but can't hurt to emphasise).